sqlglot/parser.py

```diff
@@ -5303,22 +5303,30 @@ def _parse_types(
                 any_token=False, tokens=(TokenType.VAR,)
             )
             if isinstance(identifier, exp.Identifier):
-                tokens = self.dialect.tokenize(identifier.sql(dialect=self.dialect))
+                tokens = self.dialect.tokenize(identifier.name)
 
                 if len(tokens) != 1:
                     self.raise_error("Unexpected identifier", self._prev)
```

Comment:
I think we may need to get rid of this condition, because you may end up with multiple tokens for some user-defined types, e.g. imagine you do `CREATE TYPE "foo bar baz"`– then tokenizing the `name of that will yield 3 tokens.

Let's instead check whether there's exactly one token and whether that token is in `TYPE_TOKENS`. If that's the case then we can set the `type_token` like you did, otherwise we just use the identifier and convert it to a UDT like we used to.

I think we also need to wrap `tokenize` in a try-except because if you have a UDT like `"foo'bar"`, the tokenization will fail due to the missing `'`. Weird edge case but yeah.